### PR TITLE
MABL-3059, MABL-3060, MABL-3061: support mabl branch and various fixes

### PR DESCRIPTION
**Pull request by bertold**
Charles Schwab raised these concerns. The changelog should list all the changes made.

Main drivers:
* [MABL-3060](https://mabl.atlassian.net/browse/MABL-3060) add support for mabl branch in the task configuration 
* [MABL-3059](https://mabl.atlassian.net/browse/MABL-3059) fix the generated JUnit report to correctly list associated test case IDs
* [MABL-3061](https://mabl.atlassian.net/browse/MABL-3061) updated the configuration UI to clearly

In addition:
* add description to all fields in the task configuration UI
* refactor serialization/deserialization methods into separate classes to allow better reuse
* remove powermock dependency (not used), updated Wiremock, Spotbugs and JaCoCo dependencies
* add a log4j.properties file to the test configuration
* update the changelog, readme and the task configuration screenshot
* add / extend unit tests including deserialization from JSON and serialization to XML
* fix an XML validation error (missing namespace) in the JUnit report

### CODE REVIEW

File: src/main/java/com/mabl/Utils.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.mabl.domain.ExecutionResult;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+import static org.apache.commons.lang.StringUtils.isEmpty;
+
+public class Utils {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```
Comment (DonMcNamara) [inline src/main/java/com/mabl/Utils.java:14]: Minor style comment. I associate all uppercase names with constant primitives, like a string or int, so it is a little surprising to me to see a class here. I think I would have named this something like `instance` or something like that. Ultimately, it doesn't matter too much. If we have prior art using uppercase, that's fine, of course.

File: src/main/java/com/mabl/test/output/JUnitReportSerializer.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import java.io.File;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class JUnitReportSerializer {
+    private static final Logger.Log LOG = Logger.getInstance(JUnitReportSerializer.class);
```
Comment (DonMcNamara) [inline src/main/java/com/mabl/test/output/JUnitReportSerializer.java:15]: Same comment here on upper case, but that might just be my personal preference.

**Review by DonMcNamara — APPROVED** [review]
:bamboo:  LGTM! :bamboo: Thanks for cleaning this up and adding tests. If you've changed any of the bamboo API surface that we touch, make sure you're testing it with a local and a remote agent. I don't think you did, but just wanted to mention it.

File: src/main/java/com/mabl/Utils.java (hunk 1 above)
Comment (bertold) [reply src/main/java/com/mabl/Utils.java:14]: No problem if this is what you prefer. I have always used uppercase for all constants following the guide from Sun.

**Pull request merged**